cmd/dagger/engine.go

```diff
@@ -91,7 +91,9 @@ func withEngine(
 		params.Interactive = interactive
 		params.InteractiveCommand = interactiveCommandParsed
 
-		params.PromptHandler = Frontend
+		if hasTTY {
+			params.PromptHandler = Frontend
+		}
```

Comment:
this isn't 100% correct, but an 80% solution. if you construct the process such that neither stdout or stderr are terminals, like `dagger ... 2>&1 | tee /tmp/output.txt`, this will not attempt to prompt even if stdin is a tty. detecting on stdin also has issues because you could pipe a response in or even a shell script. 

leaving PromptHandler as nil if there's no terminal to output to is a good 80% solution here, imo. the error that comes out of this is pretty explicit about what flag you need to pass, and doing this way will never leave the user in an infinite-hang-when-i-run-this-remotely situation.